libtransmission/subprocess-posix.cc

```diff
@@ -29,15 +29,18 @@ namespace
 {
 void handle_sigchld(int /*i*/)
 {
-    int rc = 0;
-
-    do
+    for (;;)
```

Comment:
It was honestly more readable as a `do-while`.
Or as:
```
int rc = 1;
while (rc > 0 || (rc == -1 && errno == EINTR))
{
    rc = waitpid(-1, nullptr, WNOHANG);
}
```

Reply:
I'll add:
>**cppcoreguidelines-avoid-do-while**
>Warns when using do-while loops. They are less readable than plain while loops, since the termination condition is at the end and the condition is not checked prior to the first iteration.

The goal of `cppcoreguidelines-avoid-do-while` was to make code more readable by having the condition BEFORE the test. By using `for (;;)`, that readability wasn't achieved: it even got worse.

Reply:
@Coeur the PR was up for a week for review before I merged it :smile: If you have readability improvements please do make a new PR :bow:

Reply:
Not sure if more readable 🤦  :
```
for (int rc = 1;
     rc > 0 || (rc == -1 && errno == EINTR);
     rc = waitpid(-1, nullptr, WNOHANG))
{
}
```